src/Helpers/JWKFetcher.php

```diff
@@ -29,22 +35,31 @@ class JWKFetcher
      *
      * @var array
      */
-    private $guzzleOptions;
+    private $httpOptions;
+
+    /**
+     * JWKS URI to use, set in $httpOptions['jwks_uri'].
+     *
+     * @var array
+     */
+    private $jwksUri;
 
     /**
      * JWKFetcher constructor.
      *
-     * @param CacheInterface|null $cache         Cache handler or null for no caching.
-     * @param array               $guzzleOptions Options for the Guzzle HTTP client.
+     * @param CacheInterface|null $cache      Cache handler or null for no caching.
+     * @param array               $httOptions HTTP options, including Guzzle options.
      */
-    public function __construct(CacheInterface $cache = null, array $guzzleOptions = [])
+    public function __construct(CacheInterface $cache = null, array $httOptions = [])
     {
         if ($cache === null) {
             $cache = new NoCacheHandler();
         }
 
-        $this->cache         = $cache;
-        $this->guzzleOptions = $guzzleOptions;
+        $this->cache       = $cache;
+        $this->httpOptions = $httOptions;
+        $this->jwksUri     = $this->httpOptions['jwks_uri'] ?? null;
```

Comment:
If you follow it through ... `getKeys()` will return empty if there is no URL to use.

Reply:
Yes! I get it. But why wait until the `getKeys()` invocation if you already know here that the URI you received is invalid and it will always fail? **Fail fast** tell the developer he set up this wrong. 

Think of the scenario in which the application gets started and because no one tried to log in so far, they never came across this issue until its too late. (This happens all the time 🗞)

Reply:
But this often wouldn't be called until the callback happens for a login 🤷‍♂️